src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -300,6 +304,10 @@ export class SavedObjectsClient {
       }
     };
 
+    if (esOptions.body.query && typeof queryDecorator === 'function') {
+      esOptions.body.query = queryDecorator(esOptions.body.query);
```

Comment:
Augments the ES query to restrict results to the active Space

Reply:
If we only support functions, let's change this to throw an Error if something else is provided instead of silently ignoring it.